BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/BlobStorage/GoogleCloudStorage/GoogleCloudStorageHealthCheck.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Text;
+using Backbone.Tooling.Extensions;
+using Google.Cloud.Storage.V1;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Backbone.BuildingBlocks.Infrastructure.Persistence.BlobStorage.GoogleCloudStorage;
+
+public class GoogleCloudStorageHealthCheck : IHealthCheck
+{
+    private const string FILE_TEXT = "healthcheck";
+    private static bool? _isHealthy;
+
+    private readonly StorageClient _storage;
+    private readonly GoogleCloudStorageOptions _options;
+
+    public GoogleCloudStorageHealthCheck(StorageClient storage, GoogleCloudStorageOptions options)
+    {
+        _storage = storage;
+        _options = options;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        if (!_isHealthy.HasValue)
```

Comment:
I just realized: if the previous health check was unsuccessful, we should give it a chance to retry. I'd say up to 5 times.